modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java

```diff
@@ -0,0 +1,172 @@
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PersistenceTestRule;
+import com.liferay.portal.test.rule.TransactionalTestRule;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+import org.frutilla.FrutillaRule;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Christian Chiappa
+ */
+@RunWith(Arquillian.class)
+public class CPDefinitionVirtualTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(), PersistenceTestRule.INSTANCE,
+			new TransactionalTestRule(
+				Propagation.REQUIRED,
+				"com.liferay.commerce.product.type.virtual.service"));
+
+	@Before
+	public void setUp() throws Exception {
+		_commerceCatalog = CommerceCatalogLocalServiceUtil.addCommerceCatalog(
+			null, RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			LocaleUtil.US.getDisplayLanguage(),
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Iterator<CPDefinition> iterator = _cpDefinitions.iterator();
+
+		while (iterator.hasNext()) {
+			CPDefinition cpDefinitionToDelete = iterator.next();
+
+			_cpDefinitionLocalService.deleteCPDefinition(
+				cpDefinitionToDelete.getCPDefinitionId());
+
+			iterator.remove();
+		}
+	}
+
+	@Test
+	public void testCreate() throws Exception {
+		frutillaRule.scenario(
+			"Add product definition"
+		).given(
+			"I add a product definition"
+		).when(
+			"ignoreSKUCombinations is true"
+		).and(
+			"hasDefaultInstance is true"
+		).and(
+			"shippable is true"
+		).then(
+			"product definition should be APPROVED"
+		).and(
+			"shippable should be false"
+		);
+
+		CPDefinition cpDefinition = CPTestUtil.addCPDefinition(
+			_commerceCatalog.getGroupId(), VirtualCPTypeConstants.NAME);
+
+		Assert.assertEquals(
+			WorkflowConstants.STATUS_APPROVED, cpDefinition.getStatus());
+
+		Assert.assertEquals("virtual", cpDefinition.getProductTypeName());
+		Assert.assertFalse(cpDefinition.isShippable());
+	}
+
+	@Test
+	public void testUpdate() throws Exception {
+		frutillaRule.scenario(
+			"Update virtual product with shippable true"
+		).given(
+			"I add a product definition with shippable false"
+		).when(
+			"shippable is now set to true"
+		).then(
+			"product definition should have shippable false"
+		);
+
+		CPDefinition cpDefinition = CPTestUtil.addCPDefinition(
+			_commerceCatalog.getGroupId(), VirtualCPTypeConstants.NAME);
+
+		long cpDefinitionId = cpDefinition.getCPDefinitionId();
+
+		Date displayDate = cpDefinition.getDisplayDate();
+		Date expirationDate = cpDefinition.getExpirationDate();
+
+		_cpDefinitionLocalService.updateCPDefinition(
```

Comment:
so you are preventing to make a virtual product to be shippable, right?

Reply:
Yes, a virtual product cannot be shipped, so it makes no sense to create a virtual product with shippable true